experiments/experiment_prototype.py

```diff
@@ -1,385 +1,1111 @@
+        that we know how to properly set up the slice and know which keys in the slice must be 
+        specified and which are unnecessary. If these keys are ever written by the user, they will 
+        be rewritten here.
+        :param exp_slice: 
+        :return: 
+        """
+
+        if 'clrfrqrange' in exp_slice.keys():
+
+            exp_slice['clrfrqflag'] = True
+            exp_slice['rxonly'] = False
+
+            txfreq = exp_slice.pop('txfreq', None)
+            if txfreq is not None and txfreq not in \
+                    range(exp_slice['clrfrqrange'][0],
+                          exp_slice['clrfrqrange'][1]):
+                pass  # TODO log a warning. Txfreq is removed but we may not be doing as you intended.
+
+            rxfreq = exp_slice.pop('rxfreq', None)
+            if rxfreq is not None and rxfreq not in \
+                    range(exp_slice['clrfrqrange'][0],
+                          exp_slice['clrfrqrange'][1]):
+                pass  # TODO log a warning. Rxfreq is removed but we may not be doing as you intended.
+
+        elif 'txfreq' in exp_slice.keys():
+            exp_slice['clrfrqflag'] = False
+            exp_slice['rxonly'] = False
+            rxfreq = exp_slice.pop('rxfreq', None)
+            if rxfreq is not None and rxfreq != exp_slice['txfreq']:
+                pass  # TODO log a warning. Rxfreq is removed but we may not be doing as you intended.
+        else:
+            exp_slice['rxonly'] = True
+            exp_slice['clrfrqflag'] = False
+
+    def check_slice_specific_requirements(self, exp_slice):
+        """
+        Check the requirements for the specific slice type as identified by the identifiers
+        rxonly and clrfrqflag. The keys that need to be checked depending on these identifiers 
+        are "txfreq", "rxfreq", and "clrfrqrange". This function may modify these keys.
+        :param exp_slice: 
+        :return: 
+        """
+        if exp_slice['clrfrqflag']:  # TX and RX mode with clear frequency search.
+            # In this mode, clrfrqrange is required along with the other requirements.
+            try:
+                assert isinstance(exp_slice['clrfrqrange'], list)
+                assert len(exp_slice['clrfrqrange']) == 2
+                assert isinstance(exp_slice['clrfrqrange'][0], int)
+                assert isinstance(exp_slice['clrfrqrange'][1], int)
+            except AssertionError:
+                errmsg = 'clrfrqrange must be an integer list of length = 2'
+                raise ExperimentException(errmsg)
+            try:
+                assert exp_slice['clrfrqrange'][0] < exp_slice['clrfrqrange'][1]
+                assert (exp_slice['clrfrqrange'][1] * 1000) < self.options.max_freq
+                assert (exp_slice['clrfrqrange'][0] * 1000) > self.options.min_freq
+            except AssertionError:
+                errmsg = """clrfrqrange must be between min and max frequencies for the radar
```

Comment:
can use .format(...) to print out the actual min and max frequencies so this err msg is more useful.